### PR TITLE
Devel service startup

### PR DESCRIPTION
**Pull request by matt9j**
This branch is based off of #10 and addresses the service file installation for freeswitch and osmocom-nitb on debian9 called out in the readme. It also fixes dependencies for restarting meas-web and smpp via runit that would sometimes fail depending on installation order.

Testing:
Tested installation and subsequent puppet agent invocations on a clean debian9 install multiple times. Ensured that freeswitch and osmocom-nitb were enabled and started successfully on reboot.

### CODE REVIEW

File: modules/rhizo_base/manifests/apt.pp [hunk 1]
```diff
@@ -12,7 +12,22 @@
 #
 
 class rhizo_base::apt {
+  # TODO Switch to "contains" once puppet3 support is no longer required.
```
Comment (whyteks) [inline modules/rhizo_base/manifests/apt.pp:15]: puppet 3 support is not required here.

File: modules/rhizo_base/manifests/apt.pp [hunk 2]
```diff
@@ -65,31 +80,35 @@
       key         => {
         'id'      => '68576280',
         'source'  => 'https://deb.nodesource.com/gpgkey/nodesource.gpg.key'
-       } 
+       }
```
Comment (whyteks) [inline modules/rhizo_base/manifests/apt.pp]: If the repo were not such a total mess already, I would ask to put WS changes or cosmetic clean ups in their own commit. Maybe going forward... :)

File: modules/rhizo_base/manifests/apt.pp (hunk 1 above)
Comment (whyteks) [reply modules/rhizo_base/manifests/apt.pp:15]: Please go ahead and do it the right way for puppet 4+

File: modules/rhizo_base/manifests/init.pp [hunk 3]
```diff
@@ -299,7 +299,7 @@
       schedule => 'always',
       ensure   => latest,
       provider => git,
-      source   => 'git@dev.rhizomatica.org:rhizomatica/rccn.git',
+      source   => 'https://github.com/Rhizomatica/rccn.git',
```
Comment (whyteks) [inline modules/rhizo_base/manifests/init.pp:306]: Ah, Yes, in this public release it should ref the public repo. Maybe we should make this a hiera parameter?

File: modules/rhizo_base/manifests/freeswitch.pp [hunk 4]
```diff
@@ -121,7 +121,7 @@
     'freeswitch-mod-syslog', 'freeswitch-mod-tone-stream',
     'freeswitch-mod-voicemail', 'freeswitch-mod-voicemail-ivr',
     'freeswitch-mod-xml-cdr', 'freeswitch-mod-cdr-pg-csv',
-    'freeswitch-sysvinit', 'libfreeswitch1',
+    'libfreeswitch1',
```
Comment (whyteks) [inline modules/rhizo_base/manifests/freeswitch.pp:139]: Good catch. Maybe related fact is that I think it was never 100% decided yet that we would move away from runit to control FS. I am in favour of using systemd

File: modules/rhizo_base/manifests/openbsc.pp [hunk 5]
```diff
@@ -53,7 +53,8 @@
   }
 
   service { 'osmocom-nitb':
-      enable  => false,
+      ensure => 'running',
```
Comment (whyteks) [inline modules/rhizo_base/manifests/openbsc.pp:56]: We would need then also to modify the systemd control file with the right switches to use the MNCC socket.
currently, nitb MUST run from runit.

**Review by whyteks — CHANGES_REQUESTED** [review]
Thanks! Great stuff. Looks like you've fixed a lot of long standing first run issues. Also, great stuff with the apt management, I never quite cracked the right way to do that without doing an apt update on each run. Please just split it up so we can merge the non problematic commits.

File: modules/rhizo_base/manifests/init.pp (hunk 3 above)
Comment (matt9j) [reply modules/rhizo_base/manifests/init.pp:306]: That makes sense. If there's a private version it should probably be parameterized. Will do!

File: modules/rhizo_base/manifests/freeswitch.pp (hunk 4 above)
Comment (matt9j) [reply modules/rhizo_base/manifests/freeswitch.pp:139]: sgtm. I know systemd is controversial, but despite people's feelings I think most can agree that having one consistent init system on a deployment, regardless of how unix like it is, is much better than having two managing different interrelated parts of the system : )

File: modules/rhizo_base/manifests/openbsc.pp (hunk 5 above)
Comment (matt9j) [reply modules/rhizo_base/manifests/openbsc.pp:56]: I am not sure if this is the case with a fresh debian9 install. The osmocom-nitb packages I believe are installing a service file? I will double check today and confirm. When I was poking around it seemed like systemctl commands were the ones controlling the nitb, and after the install even before this change systemctl status was indicating to me that the nitb is running. There may be an interaction between runit and systemd here because the osmocom-nitb service is operating in legacy sysvinit support mode. I will investigate and clarify here.

**Comment by matt9j** [comment]
Will do! I'll break things down into smaller and more focused PRs and investigate the problematic commits too.

File: modules/rhizo_base/manifests/apt.pp (hunk 1 above)
Comment (matt9j) [reply modules/rhizo_base/manifests/apt.pp:15]: Will do!

**Comment by matt9j** [comment]
Closing this PR for now to break down into smaller independent pieces.

File: modules/rhizo_base/manifests/freeswitch.pp (hunk 4 above)
Comment (whyteks) [reply modules/rhizo_base/manifests/freeswitch.pp:139]: I won't go into controversy here, but this is not about systemd. 
I don't really see runit and systemd as in the same ball park. Maybe I don't want to have runit manage my entire system, nor would I use pm2 or such either. but for keeping something up. it's great. and stood the test of time. 

TL;DR runit does a thing and it does it well.

File: modules/rhizo_base/manifests/openbsc.pp (hunk 5 above)
Comment (whyteks) [reply modules/rhizo_base/manifests/openbsc.pp:56]: What's not the case? I assure you, osmo-nitb task will NOT use the MNCC socket out of the box. the service file needs to be modified and the command line param added.

Yes, there is a mess between a sysvinit task osmo-nitb and the osmocom-nitb service (or vice versa I don't really remember. 
Just use runit. It's easier. :)
